### PR TITLE
Plat 13077 api client v2

### PR DESCRIPTION
**Pull request by taylor-cb**
## Ticket

Plat-13077
_Related tickets:_
_Related PRs:_

## Type of PR

- [ ] Bugfix
- [ ] New feature
- [X] Minor changes

Did you make changes to modules or created a new module?

- [ ] Yes, and have read the [modules updates checklist](https://github.com/crowdbotics/modules/#modules-updates-checklist) documentation and followed the instructions.
- [ ] No.

## Changes introduced

- Adding config command, which allows the user to manage their config settings via the CLI.
- Abstracting api calls to an api client
- Allowing configurable host path for api

## Test and review

_Describe how a reviewer should test your PR_

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -191,6 +192,42 @@ demo`;
...
   },
+  config: () => {
+    const args = arg({});
+
+    const action = args._[1];
+    const key = args._[2];
+    const value = args._[3];
+
+    if (!action.length) {
+      return invalid("please provide the action to perform on the config");
```
Comment (Metalheart88) [inline index.js:203]: super nit, to be consistent with the rest of the invalid messages in this file
```suggestion
      return invalid("Please provide the action to perform on the config.");
```

File: index.js [hunk 2]
```diff
@@ -191,6 +192,42 @@ demo`;
...
+    }
+
+    switch (action) {
+      case "set":
+        if (!key) {
+          return invalid("Please specific the config key to set.");
+        }
+        if (!value) {
+          return invalid("Please specific the config value to set.");
```
Comment (Metalheart88) [inline index.js:212]: should it say `specify`?
```suggestion
          return invalid("Please specify the config key to set.");
        }
        if (!value) {
          return invalid("Please specify the config value to set.");
```

File: index.js [hunk 3]
```diff
@@ -191,6 +192,42 @@ demo`;
...
+        }
+
+        configFile.set(key, value);
+        configFile.save();
+
+        break;
+      case "get":
+        if (!key) {
+          return invalid("Please specific the config key to get.");
```
Comment (Metalheart88) [inline index.js:221]: same here
```suggestion
          return invalid("Please specify the config key to get.");
```

File: index.js [hunk 4]
```diff
@@ -191,6 +192,42 @@ demo`;
...
+        }
+
+        section(configFile.get(key));
+
+        break;
+      default:
+        invalid(`invalid action "${action}" for config command`);
```
Comment (Metalheart88) [inline index.js:228]:
```suggestion
        invalid(`Invalid action "${action}" for config command.`);
```

**Review by Metalheart88 — APPROVED** [review]

**Review by alaisgomes — APPROVED** [review]

**Pull request merged**